### ISSUE TITLE
Marble diagrams for display in inline JsDocs are missing from npm package

### ISSUE THREAD

**Issue by anno1985** [ReactiveX/rxjs#5978]
## Bug Report

**Current Behavior**
The inline JsDoc/TypeDoc often make reference to images using relative paths, but those images aren't part of the npm bundle (for as far as I can tell), and thus the Inline docs, when rendered in my IDE, are missing those images.

For example, see the docs for [combineLatest](https://github.com/ReactiveX/rxjs/blob/master/src/internal/observable/combineLatest.ts#L339):
```
/**
 * Combines multiple Observables to create an Observable whose values are
 * calculated from the latest values of each of its input Observables.
 *
 * <span class="informal">Whenever any input Observable emits a value, it
 * computes a formula using the latest values from all the inputs, then emits
 * the output of that formula.</span>
 *
 * ![](combineLatest.png)
```

**Expected behavior**
I'd expect the images to be displayed. 

**Reproduction**
For example, in IntelliJ, open a rxjs ts file (e.g. `node_modules/rxjs/src/internal/observable/combineLatest.ts`), click into a method definition (with extensive comment that makes use of an image, e.g. around line 226 there), and choose View > Quick Documentation.

**Please provide repro code does not involves framework, or other setups to address this bug is specific to rxjs core**
n/a

**Environment**
n/a

**Possible Solution**

Not sure how this can be achieved with npm, but the images could potentially be shipped in a way that they don't get redistributed to end users and are available in IDEs only. Alternatively, URLs to the rxjs webserver could potentially be used (instead of relative paths).

**Additional context/Screenshots**
![inline-docs-images-missing](https://user-images.githubusercontent.com/47760138/105729461-88e7ed00-5f2d-11eb-861e-c9d42c0f89a5.png)

**Comment by benlesh** [comment]
Hmm... These should probably link to web urls. Because our npm package is already enormous.

**Comment by jakovljevic-mladen** [comment]
I don't think we can easily achieve this with our current docs generation setup. Basically, what we do right now is that we extract JSDoc comments from exported items, parse it and in case of images, [we replace such lines with the real images](https://github.com/ReactiveX/rxjs/blob/7d3c4ec727dd1962275c6f959060fdf6bdc0c164/docs_app/tools/transforms/angular-api-package/post-processors/embedMarbleDiagrams.js).

So, in our code, if we'd replace e.g.

```
![](combineLatest.png)
```

with

```
![](https://rxjs.dev/assets/images/marble-diagrams/combineLatest.png)
```

we'd have to update our `embedMarbleDiagrams.js` processor.

But, we'd have to let everybody know how to embed new or replace existing images in potential new PRs because having absolute URLs would have to be stripped in the `embedMarbleDiagrams.js` and then only the last URL path (in this case: `combineLatest.png`) would have to be used so we can embed the image properly. It's not impossible, but given the time we have on this project, I don't think we could achieve it in a timely manner.

Also, our marble diagrams need to go through massive face-lifting (we also have a new way of writing marble diagrams by using SVG images instead of PNG images that we used before).

@niklas-wortmann, what do you think about this? Would using absolute paths (like this one: `![](https://rxjs.dev/assets/images/marble-diagrams/combineLatest.png)`) create issues to maintainers that would like to update/create new images for existing operators - they'd have to know that they'd have to use absolute URL, but only the file name would be used to inject the image properly.

@anno1985, I'm open for suggestions or possibly a PR that would address this issue properly (with both `embedMarbleDiagrams.js` fix and a guide how to include existing images to the docs with the absolute URL).

**Comment by jakovljevic-mladen** [comment]
Yes, I found this issue as well:
![image](https://user-images.githubusercontent.com/28087049/217825600-b8d81479-e7cf-4984-affc-36efddc43d7a.png)

It could've been solved with migrating to SVG images which are usually much smaller:
![image](https://user-images.githubusercontent.com/28087049/217825711-c6473bd6-faf4-4ee2-b9f5-1d8737ac2ea2.png)

_(IDE that I use has a bug and it only displays black images, but this is how it would look like ☝️)._

But that's a lot of work that we need to invest. And everyone would have to know how to write a proper syntax. Also, we'd have to keep all old images in our repository for anyone with the old versions of the library to be able to access those images from our site.

**Comment by demensky** [comment]
It's black because the IDE doesn't support all sorts of colors as `rgba` and `hsl`. If replaced with HEX (like `#00aaff`) it should work.

**Comment by demensky** [comment]
Here I [have developments with diagrams](https://github.com/timdp/swirly/pull/41) that:

- [x] Drawing text with `<path />`. No need to worry about the presence of the font.
- [x] Dynamic calculation of stream title width.
- [x] Dynamic calculation of operator width. `<foreignObject />` is not used.
- [x] Lack of sub pixel coordinates. Clearer display.
- [x] Rendering without browser APIs. Can be easily run in Node.js.
- [x] `complete` & `error` now have the same radius as `next`. But `complete` & `error` become larger if `next` overlaps them. [Just like here](https://rxjs.dev/assets/images/marble-diagrams/delayWhen.png).
- [x] Built-in **dark theme**!

![diagram](https://user-images.githubusercontent.com/10235949/217831206-6e109f1e-a097-479e-8954-6d6843bceb25.svg)

Is this topic interesting?
